src/lowered.jl

```diff
@@ -44,17 +45,21 @@ function is_some_include(@nospecialize(f))
 end
 
 # This is not generally used, see `is_method_or_eval` instead
-function hastrackedexpr(stmt; heads=LoweredCodeUtils.trackedheads)
+function hastrackedexpr(stmt, code; heads=LoweredCodeUtils.trackedheads)
     haseval = false
     if isa(stmt, Expr)
         haseval = matches_eval(stmt)
         if stmt.head === :call
             f = stmt.args[1]
+            while isa(f, Core.SSAValue) || isa(f, JuliaInterpreter.SSAValue)
+                f = code[f.id]
+            end
```

Comment:
We have quite a bit of similar code in JuliaInterpreter and LoweredCodeUtils, so it might be a good idea to consolidate it into a single utility.  
Other code typically doesn’t use `while` like this but instead only follows a single level of `SSAValue` references. I’ll handle the refactoring later.

Reply:
There's lots of hacks all over the place. I think we need to take a much wider look at refactoring this stack next year, probably once JuliaLowering is ready. It's just all way too ad hoc and fragile

Reply:
Yeah overhauling the whole stack at once with integrating JuliaLowering would be very reasonable.

Reply:
Yes, once JuliaLowering is available we can make everything sane.